platform/darwin/src/NSPredicate+MGLAdditions.h

```diff
@@ -0,0 +1,108 @@
+#import <Foundation/Foundation.h>
+
+#import "NSExpression+MGLAdditions.h"
+#include <mbgl/style/filter.hpp>
+
+@interface NSPredicate (MGLAdditions)
+
+- (mbgl::style::Filter)mgl_filter;
+
+@end
+
+namespace mbgl {
+namespace darwin {
+
+class FilterEvaluator {
+public:
+
+mbgl::style::Filter operator()(NSComparisonPredicate *predicate) {
+    switch (predicate.predicateOperatorType) {
+        case NSMatchesPredicateOperatorType:
+        case NSLikePredicateOperatorType:
+        case NSBeginsWithPredicateOperatorType:
+        case NSEndsWithPredicateOperatorType:
+        case NSCustomSelectorPredicateOperatorType:
+        case NSContainsPredicateOperatorType:
+        case NSBetweenPredicateOperatorType:
```

Comment:
These are all operators we don't support. They should not be conflated with the equals operator. Instead of falling through, raise an exception explaining that the operator is unsupported.